lib/raidenclient/RaidenClient.ts

```diff
@@ -165,26 +165,28 @@ class RaidenClient extends SwapClient {
   public sendPayment = async (deal: SwapDeal): Promise<string> => {
     assert(deal.state === SwapState.Active);
     assert(deal.destination);
-    let amount = 0;
-    let tokenAddress;
+    let amount: number;
+    let tokenAddress: string;
+    let lock_timeout: number | undefined;
     if (deal.role === SwapRole.Maker) {
       // we are the maker paying the taker
       amount = deal.takerUnits;
-      tokenAddress = this.tokenAddresses.get(deal.takerCurrency);
+      tokenAddress = this.tokenAddresses.get(deal.takerCurrency)!;
```

Comment:
I think we also need to use `lock_timout=deal.takerMaxTimeLock` here to ensure that the maker's payment duration.

Reply:
I don't think so, because we're using `lock_timeout` as the delay extended to the recipient of the payment, whereas the `takerMaxTimeLock` is an upper bound on the total delay for the route. Also when the maker to taker payment is raiden, we use a hardcoded value for the route total time lock of 101 (which is then set to `takerMaxTimelock`) so I don't think it would be of much use here anyway.

Still, this is a good point to consider going forward, if we're going to be using multihop raiden payments with variable total time locks then we need to know what the maximum time lock will be for properly executing the swap.

Reply:
Ah, right - makes sense. I forgot that we're only supporting direct channels for now.

Reply:
> Still, this is a good point to consider going forward, if we're going to be using multihop raiden payments with variable total time locks then we need to know what the maximum time lock will be for properly executing the swap.

Please open an issue for this with https://github.com/ExchangeUnion/xud/issues/1068 as dependency @sangaman